util/generate_cids/generate_concept_dicts.py

```diff
@@ -94,91 +404,210 @@ def camel_case(s):
+        raise ValueError(f"Multiple zip files found in {path / version}")
+
+    # Create the output directory (if it doesn't already exist)
+    cid_dir = path / cid_folder
+    cid_dir.mkdir(parents=True, exist_ok=True)
+    LOGGER.debug(f"  Extracting FHIR JSON files to {cid_dir}")
+    with zipfile.ZipFile(files[0]) as z:
+        # Forcibly flatten the ZIP contents into the `cid_folder`
+        for zip_path in (Path(x) for x in z.namelist()):
+            with open(cid_dir / zip_path.name, "wb") as f:
+                f.write(z.read(str(zip_path)))
+
+
+def extract_table_data(
+    path: Path, version: str, ext: tuple[str, ...] = (".htm", ".html")
+) -> list[bytes]:
+    """Extract the table data from the downloaded HTML files.
+
+    Parameters
+    ----------
+    path : pathlib.Path
+        The base directory.
+    version : str
+        The name of the version subdirectory containing the HTML files.
+    ext : str, optional
+        The extension of the downloaded HTML files.
+    """
+    files = [p for p in (path / version).iterdir() if p.suffix in ext]
+    if len(files) != 2:
+        raise ValueError(
+            f"The Part 16, Chapter D and O HTML files were not found in {path / version}"
+        )
 
-def _parse_html(content):
-    # from lxml import html
-    # doc = html.document_fromstring(content)
-    return ET.fromstring(content, parser=ET.XMLParser(encoding="utf-8"))
+    # Chapter D
+    # <html xmlns="http://www.w3.org/1999/xhtml">
+    #    <head>
+    #       <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+    #       <title>D DICOM Controlled Terminology Definitions (Normative)</title>
+
+    # Chapter O
+    # <html xmlns="http://www.w3.org/1999/xhtml">
+    #    <head>
+    #       <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+    #       <title>O SNOMED Concept ID to SNOMED ID Mapping</title>
+
+    data = [None, None]
+
+    for html_file in files:
+        with open(html_file, "rb") as f:
+            file_data = f.read()
+            root = ET.fromstring(file_data, parser=ET.XMLParser(encoding="utf-8"))
+            for element in root.iter():
+                if element.tag.endswith("title"):
+                    title = element.text
+                    if title.startswith("D"):
+                        data[0] = file_data
+                    elif title.startswith("O"):
+                        data[1] = file_data
+
+                    break
+
+    if None in data:
+        raise ValueError("One or both HTML files do not contain the expected data")
 
+    return data
 
-def _download_html(url):
-    response = urllib_request.urlopen(url)
-    return response.read()
 
+def _get_text(element) -> str:
+    return "".join(element.itertext()).strip()
 
-def _get_text(element):
-    text = "".join(element.itertext())
-    return text.strip()
 
+def get_table_o1(data: bytes) -> list[tuple[str, str, str]]:
+    """Return a list of SNOMED-CT to SNOMED RT mappings.
 
```

Comment:
Pylance in VScode gives a type warning about `data` below, maybe would be less confusing to avoid re-using the name as a `list`?  I don't care too much though since this is an internal script, so to speak.

Reply:
Changed